### PR TITLE
Lock down route registration for artefacts

### PR DESCRIPTION
**Pull request by tijmenb**
Currently, artefacts owned by any app except `whitehall` can have their routes registered from this app.

This is dangerous because there 1) is a lot of old content in this database, 2) some of the routes for the old artefacts have since been registered "properly" via the publishing-api, with different routes.

There's a chance that panopticon will start sending the routes for one of these artefacts, overriding the correct routes set by the publishing-api.

This locks down the artefacts for which we'll register routes. The six are:
- business-support-finder
- calculators
- calendars
- license-finder
- publisher
- smart-answers

These are the only apps that currently rely on panopticon for route registration.

### CODE REVIEW

File: app/models/routable_artefact.rb [hunk 1]
```diff
@@ -96,6 +98,18 @@ def commit
 
   private
 
+
+  def allowed_to_register_routes_here_even_though_it_should_use_the_publishing_api?
+    %w[
+      business-support-finder
+      calculators
+      calendars
+      license-finder
```
Comment (jennyd) [inline app/models/routable_artefact.rb]: ...and this one is called `licencefinder`.

File: app/models/routable_artefact.rb [hunk 2]
```diff
@@ -96,6 +98,18 @@ def commit
 
   private
 
+
+  def allowed_to_register_routes_here_even_though_it_should_use_the_publishing_api?
+    %w[
+      business-support-finder
```
Comment (jennyd) [inline app/models/routable_artefact.rb]: This owning app is called `businesssupportfinder`.

File: app/models/routable_artefact.rb [hunk 3]
```diff
@@ -96,6 +98,18 @@ def commit
...
+      calculators
+      calendars
+      license-finder
+      publisher
+      smart-answers
```
Comment (jennyd) [inline app/models/routable_artefact.rb]: ...and this one is `smartanswers`

**Review by jennyd — CHANGES_REQUESTED** [review]
We also need to stop [Frontend registering `/help` and `/help.json` with Panopticon](https://github.com/alphagov/frontend/blob/3332f4c9e8e154e7ed4814a1ca58bd34ed11edd7/lib/tasks/panopticon.rake) - the former is [already a special route](https://github.com/alphagov/frontend/blob/3332f4c9e8e154e7ed4814a1ca58bd34ed11edd7/lib/tasks/publishing_api.rake#L36-L40) but the latter isn't.

I don't know if we need this JSON route in particular - Frontend [redirects it to `/api/help.json`](https://github.com/alphagov/frontend/blob/3332f4c9e8e154e7ed4814a1ca58bd34ed11edd7/app/controllers/help_controller.rb#L11) and then that's [handled by content API](https://github.com/alphagov/govuk-puppet/blob/66187c6b5b50191856c67da557214ded5f0e749d/modules/govuk/templates/publicapi_nginx_extra_config.erb#L48-L60).

**Comment by tijmenb** [comment]
Ouch, the app name bug is nasty. I'll have a think on how to prevent that.

**Comment by tijmenb** [comment]
@jennyd https://github.com/alphagov/panopticon/pull/404/commits/2fed5e44f75307c422a2db1b339b0b7940259826 fixes the app name issue and will prevent it from reoccurring.

**Comment by tijmenb** [comment]
@jennyd I've updated the PR to get rid of more magic strings. Decided to use constants instead of the method for readability. I haven't replaced the strings in the tests because I think that adds unnecessary noise here.

**Comment by jennyd** [comment]
👍

**Pull request merged**